### PR TITLE
(refactor): changing user struct vars from points to vars

### PR DESCRIPTION
**Pull request by Sanjay1611**
Signed-off-by: Sanjay Nathani <sanjay.nathani@mayadata.io>

This PR does followinng changes -

- Changes user struct vars from pointers to non pointers
- Changing function names in `server.go`

### CODE REVIEW

**Review by vharsh — COMMENTED** [review]
I still need to relook server.go & user.go

File: pkg/oauth/providers/github.go [hunk 1]
```diff
@@ -31,18 +30,18 @@ func getUserFromToken(c *gin.Context, token *oauth2.Token) (*models.UserCredenti
 
 	currTime := time.Now()
 	user := models.UserCredentials{
-		Name:         githubUser.Name,
+		Name:         githubUser.GetName(),
 		Kind:         models.GithubAuth,
 		Role:         models.RoleUser,
 		State:        models.StateActive,
 		SocialAuthID: githubUser.ID,
```
[the remark this answers is not in the document]
Comment (Sanjay1611) [reply pkg/oauth/providers/github.go:37]: I have left this field as pointer only because in your PR it will be changed to a string...right?

[the remark this answers is not in the document]
Comment (vharsh) [reply pkg/oauth/providers/github.go:37]: Okay, I'll change it to `string` from a `*string`

**Review by vharsh — APPROVED** [review]

File: manager/usermanager/createUser.go [hunk 2]
```diff
@@ -22,29 +20,25 @@ func CreateUser(userStore *store.UserStore, user *models.UserCredentials) (*mode
...
+			UID:             uuid.Must(uuid.NewRandom()).String(),
 			UserName:        user.UserName,
-			Password:        &password,
+			Password:        string(hashedPassword),
 			Name:            user.Name,
 			UnverifiedEmail: user.UserName,
 			Kind:            models.LocalAuth,
 			Role:            models.RoleUser,
 			OnBoardingState: models.BoardingStateSignup,
-			CreatedAt:       &time.Time{},
```
Comment (avishnu) [inline manager/usermanager/createUser.go:47]: Why is this removed?

File: pkg/oauth/providers/github.go [hunk 3]
```diff
@@ -31,18 +30,18 @@ func getUserFromToken(c *gin.Context, token *oauth2.Token) (*models.UserCredenti
 
 	currTime := time.Now()
 	user := models.UserCredentials{
-		Name:         githubUser.Name,
+		Name:         githubUser.GetName(),
```
[the remark this answers is not in the document]
Comment (avishnu) [reply pkg/oauth/providers/github.go:33]: How was this working earlier then?

File: pkg/oauth/providers/github.go [hunk 4]
```diff
@@ -31,18 +30,18 @@ func getUserFromToken(c *gin.Context, token *oauth2.Token) (*models.UserCredenti
...
-		LoggedIn:     &types.TrueValue,
+		LoggedIn:     true,
 		CreatedAt:    &currTime,
 	}
 
 	for _, githubUserEmail := range githubUserEmails {
-		if *githubUserEmail.Primary {
-			user.Email = githubUserEmail.Email
+		if githubUserEmail.Primary != nil && githubUserEmail.Email != nil {
+			user.Email = githubUserEmail.GetEmail()
```
Comment (avishnu) [inline pkg/oauth/providers/github.go:44]: Why this change?

File: manager/usermanager/createUser.go (hunk 2 above)
Comment (Sanjay1611) [reply manager/usermanager/createUser.go:47]: We are having `CreatedAt` field being set at the time of inserting the user https://github.com/mayadata-io/kubera-auth/blob/master/pkg/store/user.go#L91.
This was used for initializing the field but as we are taking the address there, it was not reqiured

File: pkg/oauth/providers/github.go (hunk 4 above)
Comment (Sanjay1611) [reply pkg/oauth/providers/github.go:44]: The package `go-github` that we are using has the `Email` field as `*string` and `user.Email` is converted to a `string`, so to get the value of email `go-github` provides this function

File: manager/usermanager/createUser.go (hunk 2 above)
Comment (avishnu) [reply manager/usermanager/createUser.go:47]: Got it !!

File: pkg/oauth/providers/github.go (hunk 4 above)
Comment (avishnu) [reply pkg/oauth/providers/github.go:44]: Got it !!

**Review by avishnu — APPROVED** [review]
lgtm
